LiteCore/Query/QueryParser.cc

```diff
@@ -1236,17 +1239,20 @@ namespace litecore {
     // Writes a call to a Fleece SQL function, including the closing ")".
     void QueryParser::writePropertyGetter(slice fn, string property, const Value *param) {
         string alias, tablePrefix;
-        if (_propertiesUseAliases) {
-            // Interpret the first component of the property as a db alias:
-            auto dot = property.find('.');
-            string rest;
-            if (dot == string::npos) {
-                dot = property.size();
-            } else {
-                rest = property.substr(dot+1);
-            }
-            alias = property.substr(0, dot);
+        // Interpret the first component of the property as an alias:
+        auto dot = property.find('.');
+        string rest;
```

Comment:
This is going to misinterpret strings like `foo[0].bar` — alias will be `foo[0]` when it should be `foo`. The code at lines 1257-59 checks for this but always rejects brackets which isn't right in general. I think you need to take the minimum position of `.` and `[`.